File: educational-core/src/com/jetbrains/edu/learning/stepik/StepikNames.java
package com.jetbrains.edu.learning.stepik;

import com.intellij.openapi.application.ApplicationManager;

public class StepikNames {
  public static final String STEPIK = "Stepik";
  public static final String STEPIK_TYPE = STEPIK;
  public static final String ARE_SOLUTIONS_UPDATED_PROPERTY = "Educational.StepikSolutionUpdated";
  public static final String STEPIK_URL = ApplicationManager.getApplication().isUnitTestMode() ? "https://release.stepik.org" : "https://stepik.org" ;
  public static final String TOKEN_URL = STEPIK_URL + "/oauth2/token/";
  public static final String STEPIK_API_URL = STEPIK_URL + "/api/";

  public static final String PYCHARM_PREFIX = "pycharm";
  public static final String EDU_STEPIK_SERVICE_NAME = "edu/stepik";
  public static final String LINK = "link";
  public static final String CLIENT_ID = StepikOAuthBundle.INSTANCE.valueOrDefault("clientId", "");
  public static final String OAUTH_SERVICE_NAME = "edu/stepik/oauth";
  public static final String EXTERNAL_REDIRECT_URL = "https://example.com";
  public static final String PYCHARM_ADDITIONAL = "PyCharm additional materials";
  public static final String ADDITIONAL_FILES = "additional_files.json";

  public static final String PLUGIN_NAME = "EduTools";
}

File: educational-core/src/com/jetbrains/edu/learning/stepik/StepikNames.kt
package com.jetbrains.edu.learning.stepik

import com.jetbrains.edu.learning.isUnitTestMode
import com.jetbrains.edu.learning.stepik.changeHost.StepikHost


object StepikNames {
  const val STEPIK = "Stepik"
  const val STEPIK_TYPE = STEPIK
  const val STEPIK_DEFAULT_URL = "https://stepik.org"
  const val STEPIK_HOST_ORDINAL_PROPERTY = "stepik.host.ordinal"
  const val STEPIK_RELEASE_URL = "https://release.stepik.org"

  const val PYCHARM_PREFIX = "pycharm"

  const val PYCHARM_ADDITIONAL = "PyCharm additional materials"
  const val ADDITIONAL_INFO = "additional_files.json"
  const val PLUGIN_NAME = "EduTools"

  @JvmStatic
  fun getStepikUrl(): String = if (isUnitTestMode) {
    STEPIK_RELEASE_URL
  }
  else {
    StepikHost.getSelectedHost().url
  }

  @JvmStatic
  fun getClientId(): String = StepikHost.getSelectedHost().clientId

  @JvmStatic
  fun getClientSecret(): String = StepikHost.getSelectedHost().clientSecret

  @JvmStatic
  fun getStepikApiUrl(): String = "${getStepikUrl()}/api/"

  @JvmStatic
  fun getTokenUrl(): String = "${getStepikUrl()}/oauth2/token/"

  @JvmStatic
  fun getStepikProfilePath(): String = "${getStepikUrl()}/users/"
}